Schema fails to compile when column is named `out`

<!--
If you want to report a bug, we added some points below you can fill out. If you want to request a feature, feel free to remove all the irrelevant text. In addition to this [issue tracker](https://github.com/diesel-rs/diesel/issues), you can also talk to Diesel maintainers and users on [Gitter](https://gitter.im/diesel-rs/diesel).
-->

## Setup

### Versions

- **Rust:** 1.35.0
- **Diesel:** 1.4.2

## Problem Description

When using `print-schema` on a database with columns named "out", the resulting schema will fail to compile with many errors like the following:

```
error[E0530]: function parameters cannot shadow unit structs
   --> src\schema.rs:287:1
    |
287 |    table! {
    |   _^
    |  |_|
    | ||
288 | ||     execution_instance (id) {
289 | ||         id -> Binary,
290 | ||         time_executed -> Nullable<Datetime>,
...   ||
306 | ||     }
307 | || }
    | || ^
    | ||_|
    | |__cannot be named the same as a unit struct
    |    the unit struct `out` is defined here
    |
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)

<!--
Thank you for your submission!  You're helping make Diesel more robust 🎉

We'll try to respond as quickly as possible.
-->

--- a/diesel/src/macros/mod.rs
+++ b/diesel/src/macros/mod.rs
@@ -23,10 +23,11 @@ macro_rules! __diesel_column {
             DB: $crate::backend::Backend,
             <$table as QuerySource>::FromClause: QueryFragment<DB>,
         {
-            fn walk_ast(&self, mut out: $crate::query_builder::AstPass<DB>) -> $crate::result::QueryResult<()> {
-                $table.from_clause().walk_ast(out.reborrow())?;
-                out.push_sql(".");
-                out.push_identifier($sql_name)
+            #[allow(non_snake_case)]
+            fn walk_ast(&self, mut __out: $crate::query_builder::AstPass<DB>) -> $crate::result::QueryResult<()> {
+                $table.from_clause().walk_ast(__out.reborrow())?;
+                __out.push_sql(".");
+                __out.push_identifier($sql_name)
             }
         }
 
@@ -808,9 +809,10 @@ macro_rules! __diesel_table_impl {
                 impl<DB: Backend> QueryFragment<DB> for star where
                     <table as QuerySource>::FromClause: QueryFragment<DB>,
                 {
-                    fn walk_ast(&self, mut out: AstPass<DB>) -> QueryResult<()> {
-                        table.from_clause().walk_ast(out.reborrow())?;
-                        out.push_sql(".*");
+                    #[allow(non_snake_case)]
+                    fn walk_ast(&self, mut __out: AstPass<DB>) -> QueryResult<()> {
+                        table.from_clause().walk_ast(__out.reborrow())?;
+                        __out.push_sql(".*");
                         Ok(())
                     }
                 }

--- a/diesel_cli/src/infer_schema_internals/inference.rs
+++ b/diesel_cli/src/infer_schema_internals/inference.rs
@@ -14,6 +14,14 @@ static RESERVED_NAMES: &[&str] = &[
     "type", "typeof", "unsafe", "unsized", "use", "virtual", "where", "while", "yield",
 ];
 
+fn is_reserved(name: &str) -> bool {
+    RESERVED_NAMES.contains(&name) || (
+        // Names ending in an underscore are not considered reserved so that we
+        // can always just append an underscore to generate an unreserved name.
+        name.starts_with("__") && !name.ends_with("_")
+    )
+}
+
 pub fn load_table_names(
     database_url: &str,
     schema_name: Option<&str>,
@@ -110,9 +118,9 @@ pub fn load_foreign_key_constraints(
     constraints.map(|mut ct| {
         ct.sort();
         ct.iter_mut().for_each(|foreign_key_constraint| {
-            if RESERVED_NAMES.contains(&foreign_key_constraint.foreign_key_rust_name.as_str()) {
+            if is_reserved(&foreign_key_constraint.foreign_key_rust_name) {
                 foreign_key_constraint.foreign_key_rust_name =
-                    format!("{}_", foreign_key_constraint.foreign_key_rust_name.as_str());
+                    format!("{}_", foreign_key_constraint.foreign_key_rust_name);
             }
         });
         ct
@@ -141,7 +149,7 @@ pub fn load_table_data(database_url: &str, name: TableName) -> Result<TableData,
     let primary_key = primary_key
         .iter()
         .map(|k| {
-            if RESERVED_NAMES.contains(&k.as_str()) {
+            if is_reserved(&k) {
                 format!("{}_", k)
             } else {
                 k.clone()
@@ -153,7 +161,7 @@ pub fn load_table_data(database_url: &str, name: TableName) -> Result<TableData,
         .into_iter()
         .map(|c| {
             let ty = determine_column_type(&c, &connection)?;
-            let rust_name = if RESERVED_NAMES.contains(&c.column_name.as_str()) {
+            let rust_name = if is_reserved(&c.column_name) {
                 Some(format!("{}_", c.column_name))
             } else {
                 None

--- a/diesel_cli/src/infer_schema_internals/inference.rs
+++ b/diesel_cli/src/infer_schema_internals/inference.rs
@@ -15,11 +15,12 @@ static RESERVED_NAMES: &[&str] = &[
 ];
 
 fn is_reserved(name: &str) -> bool {
-    RESERVED_NAMES.contains(&name) || (
-        // Names ending in an underscore are not considered reserved so that we
-        // can always just append an underscore to generate an unreserved name.
-        name.starts_with("__") && !name.ends_with("_")
-    )
+    RESERVED_NAMES.contains(&name)
+        || (
+            // Names ending in an underscore are not considered reserved so that we
+            // can always just append an underscore to generate an unreserved name.
+            name.starts_with("__") && !name.ends_with('_')
+        )
 }
 
 pub fn load_table_names(

